test/Microsoft.Health.Fhir.Tests.E2E/Rest/CorsTests.cs

```diff
@@ -0,0 +1,56 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.E2E.Common;
+using Microsoft.Health.Fhir.Web;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+using HttpMethod = System.Net.Http.HttpMethod;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest
+{
+    public class CorsTests : IClassFixture<HttpIntegrationTestFixture<Startup>>
+    {
+        private readonly FhirClient _client;
+
+        public CorsTests(HttpIntegrationTestFixture<Startup> fixture)
+        {
+            _client = fixture.FhirClient;
+        }
+
+        [Fact]
```

Comment:
Should we add a few tests to make sure the CORS are set correctly and the service accepts/rejects calls?

Reply:
The server won't reject calls based on CORS. There's some checking already in the SMART on FHIR proxy tests and this test just checks that headers come back as specified.

Currently this test is based on a permissive setup, but we could restrict it. If we restrict it we could do some negative checks in addition to a positive check.

Reply:
Also, we could add some unit tests to make sure that the CORS policy that is set is correct based on the configuration.